core/src/test/java/com/linecorp/armeria/internal/ArmeriaHttpUtilTest.java

```diff
@@ -48,7 +53,38 @@ public void testConcatPaths() throws Exception {
     }
 
     @Test
-    public void testCookieSplit() {
+    public void outboundCookiesMustBeMergedForHttp1() throws Http2Exception {
+        final HttpHeaders in = new DefaultHttpHeaders();
+
+        in.add(HttpHeaderNames.COOKIE, "a=b; c=d");
+        in.add(HttpHeaderNames.COOKIE, "e=f;g=h");
+        in.add(HttpHeaderNames.COOKIE, "i=j");
+        in.add(HttpHeaderNames.COOKIE, "k=l;");
+
+        final io.netty.handler.codec.http.HttpHeaders out =
+                new io.netty.handler.codec.http.DefaultHttpHeaders();
+
+        toNettyHttp1(0, in, out, HttpVersion.HTTP_1_1, false, true);
+        assertThat(out.getAll(HttpHeaderNames.COOKIE))
+                .containsExactly("a=b; c=d; e=f; g=h; i=j; k=l");
+    }
+
+    @Test
+    public void outboundCookiesMustBeSplitForHttp2() {
+        final HttpHeaders in = new DefaultHttpHeaders();
+
+        in.add(HttpHeaderNames.COOKIE, "a=b; c=d");
+        in.add(HttpHeaderNames.COOKIE, "e=f;g=h");
+        in.add(HttpHeaderNames.COOKIE, "i=j");
+        in.add(HttpHeaderNames.COOKIE, "k=l;");
+
+        final Http2Headers out = toNettyHttp2(in);
+        assertThat(out.getAll(HttpHeaderNames.COOKIE))
+                .containsExactly("a=b", "c=d", "e=f", "g=h", "i=j", "k=l");
+    }
+
+    @Test
+    public void inboundCookiesMustBeMerged() {
```

Comment:
Neet test for http2 input as well.

Also I think you take special care to handle Cookie, Cookie, Content-Type, Cookie etc (mixed up headers) would be good to add that to the tests

Reply:
Indeed. Fixed as suggested.